ui <-   fluidPage(
            HTML('<meta name="viewport" content="width=1024">'),
            dashboardPage(skin="black",
                    dashboardHeader(title = "Comparative Whole-blood Transcriptomics in Severe Malaria",
                                    titleWidth = 580),
                    dashboardSidebar(
                      width = 200,
                      sidebarMenu(
                        menuItem("Differential Gene Expr.", tabName = "degs", icon = icon("th")),
                        menuItem("Modular Analyses", tabName = "modularanalysis", icon = icon("cogs")),
                        menuItem("WGCNA", tabName = "wgcna", icon = icon("cogs")),
                        menuItem("Download Data", tabName = "downloads", icon = icon("download")),
                        menuItem("References", tabName = "refs", icon = icon("book"))
                      )
                    ),
                    dashboardBody(
                      tabItems(
                        tabItem(tabName = "degs",
                                h2("Differential gene expression analysis"),
                                fluidRow(
                                      box(title = "Note", status = NULL, solidHeader = F,
                                      "Differential gene expression analysis was determined using the empirical Bayes moderated T-test in the limma package. The design matrix was set to adjust for presence of sickle cell disease (HbSS genotype), HRP2 levels, and batch effects.",
                                      br(), width = 12),
                                      column(width = 12,   
                                         box(title = "Significance Thresholds", status = NULL, solidHeader = F,
                                             sliderInput("degfdr",
                                                         "FDR (adjusted P value) for determining numbers of up- or down-regulated genes:",
                                                         min = 0,
                                                         max = 100,
                                                         value = 10,
                                                         ticks = TRUE,
                                                         step = 5,
                                                         width = 225),
                                             sliderInput("degfc",
                                                         "log2 fold change threshold for determining numbers of up- or down-regulated genes:",
                                                         min = 0,
                                                         max = 4,
                                                         value = 0.585,
                                                         round = FALSE,
                                                         step = 0.005,
                                                         width = 225),
                                             checkboxInput("genesymbolsonly",
                                                         "Filter out transcripts without gene symbols",
                                                         width = 225),
                                             checkboxInput("duplicated",
                                                           "If gene duplicated, retain only most significant target",
                                                           width = 225),
                                             downloadButton("exportTable1", label = "Download all filtered DEGs"),
                                             downloadButton("exportTable2", label = "Download CMvsCC filtered DEGs"),
                                             downloadButton("exportTable3", label = "Download SMAvsCC filtered DEGs"),
                                             downloadButton("exportTable4", label = "Download CMvsSMA filtered DEGs"),
                                             width = 2),
                                         box(title = "Cerebral Malaria (CM) vs Community Children (CC)", status = NULL, solidHeader = F,
                                             collapsible = TRUE, collapsed = F,
                                             DT::dataTableOutput("table2"),
                                             width = 10, align = "right"),
                                         box(title = "Severe Malarial Anemia (SMA) vs Community Children (CC)", status = NULL, solidHeader = F,
                                             collapsible = TRUE, collapsed = F,
                                             DT::dataTableOutput("table3"),
                                             width = 10, align = "right"),
                                         box(title = "Cerebral Malaria (CM) vs Severe Malarial Anemia (SMA)", status = NULL, solidHeader = F,
                                             collapsible = TRUE, collapsed = F,
                                             DT::dataTableOutput("table4"),
                                             width = 10, align = "right")))),
                        tabItem(tabName = "modularanalysis",
                          h2("Modular Analysis"),
                          fluidRow(
                              box(title = "Instructions", status = NULL, solidHeader = F,
                                     "The tmod Panel Plot shows a heatmap-like plot. Each row corresponds to one module, and columns correspond to the two-way comparisons for which the tmod analyses were run. Each significantly enriched module is shown in red (upregulated in first group) or blue (upregulated in second group). Size of the circle/box corresponds to the effect size (for example, AUC in the CERNO test), and intensity of the color corresponds to the q-value.",
                                     br(),
                                     br(),
                                     "Adjusting q-value changes the significance threshold for modules displayed.",
                                     br(),
                                     br(),
                                     "Adjusting the false discovery rate and fold-change thresholds changes the thresholds for calculating the numbers of significantly up- and down-regulate genes for each module.", br(),
                                  width = 12
                                 ),
                              column(width = 12),
                              box(title = "Controls", status = NULL, solidHeader = F,
                              selectInput("MSET",
                                              "Module Set:",
                                              c("Li" = "LI", "Chaussabel" = "DC", "Both" = "all"), "LI",
                                              width = 225),
                              selectInput("PieType",
                                              "Plot type:",
                                              c("rug", "boxpie", "pie"), "rug",
                                              width = 225),
                              sliderInput("qval",
                                          "q-value threshold for module significance:",
                                          min = 1,
                                          max = 25,
                                          value = 5,
                                          ticks = FALSE,
                                          step = 1,
                                          width = 225),
                              sliderInput("fdr",
                                          "FDR for determining numbers of up- or down-regulated genes:",
                                          min = 1,
                                          max = 25,
                                          value = 25,
                                          ticks = FALSE,
                                          step = 1,
                                          width = 225),
                              sliderInput("fc",
                                          "log2 fold change threshold for determining numbers of up- or down-regulated genes:",
                                          min = 0,
                                          max = 2,
                                          value = 0,
                                          ticks = TRUE,
                                          step = 0.5,
                                          width = 225),
                             downloadButton("export", label = "Download Tmod plot pdf"),
                              width = 2
                              ),
                              box(title = "Tmod Panel Plot",  status = NULL, solidHeader = F,
                                  collapsible = FALSE,
                                  plotOutput("distPlot"),
                                  height = 1600, width = 10, align = "right"))),
                        tabItem(tabName = "wgcna",
                                h2("Weighted gene Co-expression Network Analysis for CMvsSMA"),
                                fluidRow(
                                  box(title = "Note", status = NULL, solidHeader = F,
                                      "The WGCNA is a comprehensive collection of R functions for performing various aspects of weighted correlation network analysis. The MDS plot shows top modules with similar gene co-expression patterns. The dendrogram represents the unsupervised heirarchichal clustering of the top modules. The module-trait relationship is a correlation plot between top modules and outcome data for CM SMA patients. The BTM plot is pathway analysis with top co-expressing genes ranked by their coefficients for all comparisons.",br(),
                                      "For more information about WGCNA package, refer to: ", a("https://www.jstatsoft.org/article/view/v046i11", href = "https://www.jstatsoft.org/article/view/v046i11"),
                                      br(), width = 12),
                                  column(width = 12,   
                                  box(title = "Modules representation", status = NULL, solidHeader = F,
                                             selectInput("MES",
                                                         "Modules plot:",
                                                         c("MDS plot" = "MDS",
                                                           "Modules Dendrogram" = "MDRO",
                                                           "Module trait plot" = "MT",
                                                           "Ranked BTM pathways" = "BTM"), "MDS",
                                                         width = 225),
                                             downloadButton("export1", label = "Download plot pdf"),
                                             width = 2
                                         ),
                                  box(title = "WGCNA plots",  status = NULL, solidHeader = F,
                                             collapsible = FALSE,
                                             plotOutput("distPlot1"),
                                             height = 1000, width = 7, align = "right")))),
                        tabItem(tabName = "refs",
                              box(title = "References",  status = NULL, solidHeader = F,
                                  "Nallandhighal et al. J Infect Dis. 2019 Jan 1;219(1):154-164.", br(),
                                  a("doi: 10.1093/infdis/jiy468", href = "https://doi.org/10.1093/infdis/jiy468", target="_blank"), br(),
                                  br(),
                                  "Analyses were performed using the tmod package:", br(),
                                  "January Weiner (2018). tmod: Feature Set Enrichment Analysis for Metabolomics and Transcriptomics. R package version 0.36.", br(),
                                  a("https://CRAN.R-project.org/package=tmod", href = "https://CRAN.R-project.org/package=tmod", target="_blank"), br(),
                                  br(),
                                  "WGCNA analysis was performed using:", br(),
                                  "Langfelder P and Horvath S, WGCNA: an R package for weighted correlation network analysis. BMC Bioinformatics 2008, 9:559",br(),
                                  a("https://CRAN.R-project.org/package=WGCNA", href= "https://CRAN.R-project.org/package=WGCNA", target="_blank"),br(),
                                  width = 12)),
                        tabItem(tabName = "downloads",
                              box(title = "Download Data",  status = NULL, solidHeader = F,
                                  "Raw and normalized expression data as well as metadata is available in the Gene Expression Omnibus through the accession number GSE117613.", br(),
                                  br(),
                                  a("Link to data", href = "https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE117613", target="_blank"), br(),
                                  br(),
                                  "Box links to download CSV data files.",br(),
                                  br(),
                                  a("Link to download Metadata", href="https://iu.box.com/s/0rtzjxt2xsek75xzwevdxwmj6sse8ye9", target="_blank"),br(),
                                  a("Link to download Normalized data", href="https://iu.box.com/s/e3osihy8kajh7r9la92mrpfd80g7ueus", target="_blank"),br(),
                                  a("Link to download Raw data", href="https://iu.box.com/s/hg7rfgvyjiuaqffpy5eydyp0dzdeu1ai", target="_blank"),br(),
                                  # downloadButton("GEOmeta", label="Download Metadata"),
                                  # downloadButton("GEOnorm", label="Download Normalized data"),
                                  # downloadButton("GEOraw", label="Download Raw data"),
                                  width = 12))))
                      ))
